converter/src/main/java/gov/cms/qpp/conversion/model/error/Detail.java

```diff
@@ -1,27 +1,60 @@
 package gov.cms.qpp.conversion.model.error;
 
+import java.io.Serializable;
+import java.util.Objects;
+
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
 
-import java.io.Serializable;
+import gov.cms.qpp.conversion.model.Node;
+import gov.cms.qpp.conversion.model.TemplateId;
 
 /**
  * Holds the error information from Validators.
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
 public class Detail implements Serializable {
 	private static final long serialVersionUID = 8818544157552590676L;
+
+	/**
+	 * Creates a mutable Detail based on the given error and node
+	 */
+	public static Detail forErrorAndNode(LocalizedError error, Node node) {
+		Objects.requireNonNull(node, "node");
+
+		Detail detail = forErrorCode(error);
+		detail.setPath(node.getPath());
+		return detail;
+	}
+
+	/**
+	 * Creates a mutable Detail based on the given error
+	 * @param code
+	 * @return
+	 */
+	public static Detail forErrorCode(LocalizedError error) {
+		Objects.requireNonNull(error, "error");
+
+		Detail detail = new Detail();
+		detail.setErrorCode(error.getErrorCode());
+		detail.setMessage(error.getMessage());
+		return detail;
+	}
+
+	@JsonProperty("errorCode")
+	private ErrorCode errorCode;
 	@JsonProperty("message")
 	private String message;
 	@JsonProperty("path")
-	private String path = "";
+	private String path;
```

Comment:
Any chance this'll result in a null pointer somewhere?

Reply:
All tests pass, and I couldn't think of when it would be.

Reply:
I think I may have been wrong. Investigating (& fixing) now.